torchgeo/datasets/bigearthnetv2.py

```diff
@@ -0,0 +1,664 @@
+        split: str = "train",
+        bands: str = "all",
+        num_classes: int = 19,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new BigEarthNet dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: train/val/test split to load
+            bands: load Sentinel-1 bands, Sentinel-2, or both. one of {s1, s2, all}
+            num_classes: number of classes to load in target. one of {19, 43}
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+        """
+        assert split in self.splits_metadata
+        assert bands in ["s1", "s2", "all"]
+        assert num_classes in [43, 19]
+        self.root = root
+        self.split = split
+        self.bands = bands
+        self.num_classes = num_classes
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        self.class2idx_43 = {c: i for i, c in enumerate(self.class_sets[43])}
+        self.class2idx_19 = {c: i for i, c in enumerate(self.class_sets[19])} 
+        # self._verify()
+        self.folders = self._load_folders()
+
+    def class2idx(self, label:str, level=19):
+        assert level == 19 or level == 43, "level must be 19 or 43"
+        return self.class2idx_19[label] if level == 19 else self.class2idx_43[label]
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        image = self._load_image(index)
+        label = self._load_target(index)
+        sample: dict[str, Tensor] = {"image": image, "label": label}
+        # sample: dict[str, Tensor] = {"image": image, "label": None}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.folders)
+
+    def _load_folders(self) -> list[dict[str, str]]:
+        """Load folder paths.
+
+        Returns:
+            list of dicts of s1 and s2 folder paths
+        """
+        filename = self.splits_metadata[self.split]["filename"]
+        dir_s1 = self.metadata_locs["s1"]["directory"]
+        dir_s2 = self.metadata_locs["s2"]["directory"]
+        dir_maps = self.metadata_locs["maps"]["directory"]
+
+        self.metadata = pd.read_parquet(os.path.join(self.root, filename))
```

Comment:
Just noticed, that this is missing a filter based on the split for the appropriate set.